building.rb

```diff
@@ -1,4 +1,30 @@
 
 class Building
 
+  attr_reader :address, :apartments
+
+  def initialize(args={})
+    @address = args.fetch(:address)
+    @apartments = args.fetch(:apartments)
+  end
+
+  def total_room_count
+    @apartments.reduce(0) { |sum, apt| sum + apt.rooms.length }
+  end
```

Comment:
Also, instead of working with istance variable directly, use getter method `apartments`.